config/kube.py

```diff
@@ -207,33 +298,63 @@ def kube_reset():
                 ignore_error=True)
 
     clicommon.run_command("kubeadm reset -f", ignore_error=True)
-    clicommon.run_command("rm -rf /etc/cni/net.d")
-    clicommon.run_command("rm -f {}".format(KUBE_ADMIN_CONF))
+    clicommon.run_command("rm -rf {}".format(CNI_NET_DIR))
+    if purge_conf:
+        clicommon.run_command("rm -f {}".format(KUBE_ADMIN_CONF))
     clicommon.run_command("systemctl stop kubelet")
     clicommon.run_command("systemctl disable kubelet")
 
+    _update_kube_server_state(state_db, False)
+
+
+def kube_reset(force=False):
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    state_db = _get_state_db()
+    if not force:
+        if not is_connected():
+            # Already *not* connected. No-Op. Just ensure SERVER-DB is updated
+            _update_kube_server_state(state_db, False)
+            return
+
+    _do_reset(state_db, True)
+
 
 def kube_join(force=False):
     lock_fd = _take_lock()
     if not lock_fd:
         log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
         return
 
-    db_data = Db().get_data('KUBERNETES_MASTER', 'SERVER')
-    if not db_data or 'IP' not in db_data or not db_data['IP']:
+    db_data = {
+            KUBE_SERVER_IP: "",
+            KUBE_SERVER_DISABLE: "false"
+            }
+    db_data.update(Db().get_data(KUBE_SERVER_TABLE_NAME, KUBE_SERVER_TABLE_KEY))
+
+    if (not db_data[KUBE_SERVER_IP]):
         log.log_error("Kubernetes server is not configured")
 
-    if db_data['disable'].lower() != "false":
+    if db_data[KUBE_SERVER_DISABLE].lower() != "false":
         log.log_error("kube join skipped as kubernetes server is marked disabled")
         return
 
+    state_db = _get_state_db()
+
     if not force:
-        if _is_connected(db_data['IP']):
-            # Already connected. No-Op
+        if is_connected(db_data[KUBE_SERVER_IP]):
+            # Already connected. No-Op. Just ensure SERVER-DB is updated
+            _update_kube_server_state(state_db, True, db_data[KUBE_SERVER_IP])
             return
 
```

Comment:
Should there be a check to see if the server is reachable?

Reply:
Same comment as above. Let me check and fix as needed.

Reply:
When server reachability was being checked on wrong port for me (originally HAProxy ran service on port 80), server reachability showed false. But when I tried to join, there was no error output

Reply:
port_check is done on demand and for now, the only demand is pending labels to be written into API server. If not reachable,  a periodic try would happen. Port check is also done for port 6443 only. 

I took your earlier comment to make it configurable. 

Let me test incorrect IP.

Reply:
This code is removed. So resolving.